Lib/test/test_float.py

```diff
@@ -701,6 +701,21 @@ def test_issue5864(self):
         self.assertEqual(format(1234.56, '.4'), '1.235e+03')
         self.assertEqual(format(12345.6, '.4'), '1.235e+04')
 
+    def test_issue35560(self):
+        self.assertEqual(format(123.0, '00'), '123.0')
+        self.assertEqual(format(123.34, '00f'), '123.340000')
+        self.assertEqual(format(123.34, '00e'), '1.233400e+02')
+        self.assertEqual(format(123.340, '00g'), '123.34')
+        self.assertEqual(format(123.340, '00.10f'), '123.3400000000')
+        self.assertEqual(format(123.340, '0010f'), '123.340000')
```

Comment:
The bug is not reproducible with non-zero width. Add instead tests for non-zero precision and other format characters:  '00.10e', '00.10g'.